xdsl/transforms/arith_add_fastmath.py

```diff
@@ -0,0 +1,90 @@
+"""Passes to manipulate fastmath flags in FP arith operations."""
+
+from dataclasses import dataclass
+from typing import Literal
+
+from xdsl.dialects import arith, builtin, llvm
+from xdsl.ir import Operation
+from xdsl.passes import MLContext, ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+)
+
+
+def _match_string_to_enum(flag: str, enum_type: type[llvm.FastMathFlag]):
+    for member in enum_type:
+        if flag == str(member.value):
+            return member
+
+    raise ValueError(f"{flag} is not a valid fastmath flag.")
+
+
+def _get_flag_list(flags: list[str], enum_type: type[llvm.FastMathFlag]):
+    list_flags: list[enum_type] = []
+    for flag in flags:
+        list_flags.append(_match_string_to_enum(flag, llvm.FastMathFlag))
+    return list_flags
```

Comment:
```suggestion
_STRING_TO_ENUM = {str(member.value): member for member in llvm.FastMathFlag}

def _get_flag_list(flags: list[str]):
    try:
        return [_STRING_TO_ENUM[flag] for flag in flags] 
    except KeyError:
        raise ValueError(f"{flag} is not a valid fastmath flag.")    
```

Reply:
Feels worth caching, and it seems like the enum_type parameter won't ever be different from `llvm.FastMathFlag`